PubNub/Misc/Helpers/PNString.m

```diff
@@ -16,27 +16,16 @@ @implementation PNString
 #pragma mark - Encoding
 
 + (NSString *)percentEscapedString:(NSString *)string {
-    
-    static NSCharacterSet *allowedCharacters;
-    static dispatch_once_t onceToken;
-    dispatch_once(&onceToken, ^{
-        
-        NSMutableCharacterSet *chars = [[NSMutableCharacterSet URLPathAllowedCharacterSet] mutableCopy];
-        [chars formUnionWithCharacterSet:[NSCharacterSet URLQueryAllowedCharacterSet]];
-        [chars formUnionWithCharacterSet:[NSCharacterSet URLFragmentAllowedCharacterSet]];
-        [chars removeCharactersInString:@":/?#[]@!$&’()*+,;="];
-        
-        allowedCharacters = [chars copy];
-    });
-    
+
     // Wrapping non-string object (it can be passed from dictionary and compiler at run-time won't notify 
     // about different data types.
     if (![string respondsToSelector:@selector(length)]) {
         
         string = [NSString stringWithFormat:@"%@", string];
     }
     // Escape unallowed characters
-    NSString *escapedString = [string stringByAddingPercentEncodingWithAllowedCharacters:allowedCharacters];
+    NSString *escapedString = (NSString *)CFBridgingRelease(CFURLCreateStringByAddingPercentEscapes(NULL, (__bridge CFStringRef)string, NULL, CFSTR(" \"#%/:<>?@[\\]^`{|}"), kCFStringEncodingUTF8));
```

Comment:
This doesn't seem to be the same set of characters as what was removed above in `removeCharactersInString`